### PR TITLE
[Merged by Bors] - feat: The norm on `Unitization` is a C⋆-norm

### PR DESCRIPTION
**Pull request by j-loreaux**
This shows that C⋆-algebras are always `RegularNormedAlgebra`s, so that their `Unitization` is equipped with a norm. Moreover, we show this norm is a C⋆-norm.

- [x] depends on: #5330
- [x] depends on: #5741
- [x] depends on: #5742 
- [x] depends on: #5743 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5330
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5330~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by dupuisf** [comment]
Would it be a lot of trouble to split this into a PR that just renames files, and the other one with the actual changes made here?

**Comment by j-loreaux** [comment]
@dupuisf I have split this into more manageable chunks. 5741 is easy and just renames instances, 5743 is easy and just renames the file, 5742 is the bulk of the work (`RegularNormedAlgebra` definition and the norm on `Unitization`), and then this PR just tops it off with the C*-algebra results.

File: Mathlib/Analysis/NormedSpace/Star/Unitization.lean [hunk 1]
```diff
@@ -4,68 +4,170 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
+/-! # The minimal unitization of a C⋆-algebra
 
-/-! # The left-regular representation is an isometry for C⋆-algebras -/
+This file shows that when `E` is a C⋆-algebra (over a densely normed field `𝕜`), that the minimal
+`Unitization` is as well. In order to ensure that the norm structure is available, we must first
+show that every C⋆-algebra is a `RegularNormedAlgebra`.
 
+In addition, we show that in a `RegularNormedAlgebra` which is a `StarRing` for which the
+involution is isometric, that multipliciation on the right is also an isometry (i.e.,
+`Isometry (ContinuousLinearMap.mul 𝕜 E).flip`).
```
Comment (dupuisf) [inline Mathlib/Analysis/NormedSpace/Star/Unitization.lean:18]:
```suggestion
involution is isometric, that multiplication on the right is also an isometry (i.e.,
`Isometry (ContinuousLinearMap.mul 𝕜 E).flip`).
```

File: Mathlib/Analysis/NormedSpace/Star/Unitization.lean [hunk 2]
```diff
@@ -4,68 +4,170 @@ Released under Apache 2.0 license as described in the file LICENSE.
...
+    -- in this step we make use of the key lemma `norm_splitMul_snd_sq`
+    have h₂ : ‖(Unitization.splitMul 𝕜 E (star x * x)).snd‖
+        = ‖(Unitization.splitMul 𝕜 E x).snd‖ ^ 2 := by
+      refine le_antisymm ?_ (norm_splitMul_snd_sq 𝕜 x)
+      rw [map_mul, Prod.snd_mul]
+      exact (norm_mul_le _ _).trans <| by
+        rw [sq]
+        gcongr
+        simpa only [star_star] using h₁ (star x)
+    -- Show that `(Unitization.splitMul 𝕜 E x).fst` satisifes the C⋆-property
```
Comment (dupuisf) [inline Mathlib/Analysis/NormedSpace/Star/Unitization.lean:160]:
```suggestion
    -- Show that `(Unitization.splitMul 𝕜 E x).fst` satisfies the C⋆-property
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Pull request closed without merging**